src/rawposix/src/net_calls.rs

```diff
@@ -0,0 +1,411 @@
+        return handle_errno(errno, "getsockopt");
+    }
+
+    ret
+}
+
+/// Reference to Linux: https://man7.org/linux/man-pages/man2/getpeername.2.html
+///
+/// The Linux `getpeername()` syscall retrieves the address of the peer connected to a socket.
+/// This implementation obtains the socket file descriptor and address buffer from the current cage,
+/// then invokes the host kernel's `getpeername()` and writes the result to user space.
+///
+/// ## Input:
+///     - cageid: identifier of the current cage
+///     - fd_arg: virtual file descriptor of the connected socket
+///     - addr_arg: pointer to a buffer in user space to store the peer address
+///
+/// ## Return:
+///     - On success: 0  
+///     - On failure: negative errno indicating the error
+pub fn getpeername_syscall(
+    cageid: u64,
+    fd_arg: u64,
+    fd_cageid: u64,
+    addr_arg: u64,
+    addr_cageid: u64,
+    arg3: u64,
+    arg3_cageid: u64,
+    arg4: u64,
+    arg4_cageid: u64,
+    arg5: u64,
+    arg5_cageid: u64,
+    arg6: u64,
+    arg6_cageid: u64,
+) -> i32 {
+    let fd = convert_fd_to_host(fd_arg, fd_cageid, cageid);
+    let addr = sc_convert_addr_to_host(addr_arg, addr_cageid, cageid);
+
+    if !(sc_unusedarg(arg3, arg3_cageid)
+    &&sc_unusedarg(arg4, arg4_cageid)
+    && sc_unusedarg(arg5, arg5_cageid)
+    && sc_unusedarg(arg6, arg6_cageid))
+    {
+        return syscall_error(Errno::EFAULT, "getpeername_syscall", "Invalide Cage ID");
+    }
+
+    let (finalsockaddr, mut addrlen) = convert_host_sockaddr(addr, addr_cageid, cageid);
+    let ret = unsafe { libc::getpeername(fd, finalsockaddr, &mut addrlen as *mut u32) };
+
+    if ret < 0 {
+        let errno = get_errno();
+        return handle_errno(errno, "getpeername");
+    }
+
+    ret
+}
+
+/// Reference to Linux: https://man7.org/linux/man-pages/man2/socketpair.2.html
+///
+/// The Linux `socketpair()` syscall creates a pair of connected sockets.
+/// This implementation creates the socket pair in the host kernel and assigns virtual file descriptors
+/// to the resulting sockets within the current cage.
+///
+/// ## Input:
+///     - cageid: identifier of the current cage
+///     - domain_arg: communication domain (e.g., AF_UNIX)
+///     - type_arg: communication semantics (e.g., SOCK_STREAM)
+///     - protocol_arg: protocol to be used
+///     - virtual_socket_vector_arg: pointer to a `SockPair` structure in user space to receive the result
+///nullity1/2
+/// ## Return:
+///     - On success: 0  
+///     - On failure: negative errno indicating the error
+pub fn socketpair_syscall(
+    cageid: u64,
+    domain_arg: u64,
+    domain_cageid: u64,
+    type_arg: u64,
+    type_cageid: u64,
+    protocol_arg: u64,
+    protocol_cageid: u64,
+    virtual_socket_vector_arg: u64,
+    virtual_socket_vector_cageid: u64,
+    arg5: u64,
+    arg5_cageid: u64,
+    arg6: u64,
+    arg6_cageid: u64,
+) -> i32 {
+    let domain = sc_convert_sysarg_to_i32(domain_arg, domain_cageid, cageid);
```

Comment:
This syscall also has flags and creates fds so need to check for cloexec at least